src/main/webapp/app/course/course-scores/scores-storage.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { CourseScoresDTO } from 'app/course/course-scores/course-scores-dto';
+import { ExerciseType, ExerciseTypeTOTAL } from 'app/entities/exercise.model';
+import { Result } from 'app/entities/result.model';
+
+@Injectable({ providedIn: 'root' })
+export class ScoresStorageService {
+    private storedScoresPerExerciseType: Map<number, Map<ExerciseType | ExerciseTypeTOTAL, CourseScoresDTO>> = new Map();
```

Comment:
Since this `Map<ExerciseType | ExerciseTypeTOTAL, CourseScoresDTO>` type is used at many places, it might makes sense to define this as an explicit type.